src/main/java/com/amazonaws/encryptionsdk/AwsCrypto.java

```diff
@@ -93,6 +95,11 @@ public class AwsCrypto {
   public AwsCrypto() {
     commitmentPolicy_ = CommitmentPolicy.ForbidEncryptAllowDecrypt;
     maxEncryptedDataKeys_ = CiphertextHeaders.NO_MAX_ENCRYPTED_DATA_KEYS;
+    LOGGER.warning(
+            "This major version (1.x) of the AWS Encryption SDK for Java has reached End-of-Support.\n" +
+            "It will no longer receive security updates or bug fixes.\n" +
+            "Consider updating to the latest version of the AWS Encryption SDK."
+    );
   }
 
   private AwsCrypto(Builder builder) {
```

Comment:
It looks like this constructor and the above are separate code paths. We should also add the warning in the `AwsCrypto(Builder builder)` constructor.

If this were HEAD I'd suggest refactoring so the public constructor uses the private one, so we'd only have to add it in one place. But since this is code we are not maintaining anymore, I'm ok with just copy-pasta-ing the warning where it's needed.

Reply:
Fixed in 6f3e2de